File: app/src/main/java/be/helmo/myscout/database/dao/MeetingDao.java
package be.helmo.myscout.database.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;
import java.util.UUID;

import be.helmo.myscout.model.Meeting;


@Dao
public interface MeetingDao {

    @Query("SELECT * FROM Meeting")
    LiveData<List<Meeting>> getMeetings();

    @Query("SELECT * FROM Meeting where id = (:uuid)")
    LiveData<Meeting> getMeeting(UUID uuid);

    @Insert
    void insert(Meeting meeting);

    @Update
    void update(Meeting meeting);
}

File: app/src/main/java/be/helmo/myscout/database/dao/MeetingDao.kt
package be.helmo.myscout.database.dao

import androidx.room.*
import be.helmo.myscout.model.Meeting
import be.helmo.myscout.model.Phase
import com.google.android.gms.maps.model.LatLng
import kotlinx.coroutines.flow.Flow
import java.util.*

@Dao
interface MeetingDao {

    @get:Query("SELECT * FROM Meeting ORDER BY startDate")
    val meetings: Flow<List<Meeting?>?>?

    @Query("SELECT id, description, story, startDate, endDate, startLocation, endLocation, rating FROM Meeting " +
            "WHERE id = (:uuid)")
    fun getMeeting(uuid: UUID?): Flow<Meeting?>?

    @Insert
    fun insert(meeting: Meeting?)

    @Update
    fun update(meeting: Meeting?)

    @Delete
    fun delete(meeting: Meeting?)

}